pxtblocks/fields/field_asset.ts

```diff
@@ -245,11 +263,13 @@ namespace pxtblockly {
                         if (this.sourceBlock_ && this.asset.meta.blockIDs) {
                             this.asset.meta.blockIDs = this.asset.meta.blockIDs.filter(id => id !== this.sourceBlock_.id);
 
-                            if (this.asset.meta.blockIDs.length === 0 && !this.asset.meta.displayName) {
-                                project.removeAsset(this.asset);
-                            }
-                            else {
-                                project.updateAsset(this.asset);
+                            if (!this.isTemporaryAsset()) {
+                                if (this.asset.meta.blockIDs.length === 0 && !this.asset.meta.displayName) {
```

Comment:
not sure i understand this check--won't the display name always be present if it is not a temporary asset?

Reply:
whoops, i think this is just left over from refactoring